src/Time.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Robotec.AI
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#ifdef RGL_BUILD_ROS2_EXTENSION
+#include <builtin_interfaces/msg/time.hpp>
+#endif
+
+struct Time
+{
+	static Time zero() { return Time(0); }
+	static Time seconds(double seconds) { return Time(static_cast<uint64_t>(seconds * 1.0e9)); }
+	static Time nanoseconds(uint64_t nanoseconds) { return Time(nanoseconds); }
```

Comment:
Can You explain why You decided to make private constructor and overlay it with few public static methods?
This is not a mistake. Just want to fully understand intention.

Reply:
I wanted to make the code cleaner when creating Time objects. Having constructors with parameters that differ on param types, there is no explicit information on what units you pass.
For example, `Time(1)` and `Time(1.0f)` will make a huge difference in what you get, and you need to check the implementation of every constructor to know the value of the time you have created.